### PR TITLE
refactor: toggleInputList tests

### LINKED ISSUE #854: Refactor ToggleInputList and ToggleBoxGroup tests from Enzyme to React Testing Library
# Description
This issue is related to refactoring the  `ToggleInputList.test.tsx`, and `ToggleBoxGroup.test.tsx`.

We've recently been migrating our tests from Jest + Enzyme to Jest + testing-library/react to support a full migration to React 18.

You may refer to this previous PR beginning the migration for code examples - https://github.com/dcos-labs/ui-kit/pull/771. Or you can find many code examples throughout other component tests using react testing library within the codebase. 

## Acceptance Criteria
Refactor the test to use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) instead of [Enzyme](https://www.npmjs.com/package/enzyme) (which is no longer supported with React 18). 

More information: [Migrate from Enzyme](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/)

## How to Contribute 
Please review the [contributing documentation](https://github.com/dcos-labs/ui-kit/blob/main/CONTRIBUTING.md) beforehand. 

Steps to Contribute:
1. [Fork the repository](https://github.com/dcos-labs/ui-kit/fork).
2. Clone it to your local system.
3. Create a new branch.
4. Make your changes. Make sure that tests are still passing. To update snapshot tests, run `npm run test -- -u` and include the updated tests in your commit.
5. Commit your changes.  When committing prefix your commit message with the appropriate conventional commit type. For example, `refactor: <component_name> tests`. 
6. Push up your changes. In GitHub, click the compare & pull request button to begin drafting your pull request.
7. Fill out the pull request form and attach the "ready for review" label when complete.

### PR DESCRIPTION
**Pull request by taksuparth**
# Description
- Migrate ToggleInputList tests from enzyme to react-testing-library
- Migrate ToggleGroupBox tests from enzyme to react-testing-library

## Which issue(s) does this PR relate to?
- Closes #854 

## Testing
- `npm run test`

## Trade-offs
- N/A

## Screenshots
- N/A

## Checklist

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [x] This PR is associated with a JIRA and mentions in the commit message footer ("Closes …")
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [x] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: packages/toggleInputList/tests/ToggleInputList.test.tsx [hunk 1]
```diff
@@ -15,78 +16,87 @@ expect.addSnapshotSerializer(createSerializer());
...
+    const { getByTestId } = render(
       <ToggleInputList
         id="checkbox"
         items={options}
         listLabel="Sample legend"
         required={true}
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    getByTestId("dangerText");
```
Comment (nataliepina) [inline packages/toggleInputList/tests/ToggleInputList.test.tsx:95]: Should there be an assertion of some sort here?

File: packages/toggleInputList/components/ToggleInputList.tsx [hunk 2]
```diff
@@ -98,7 +98,10 @@ const ToggleInputList = ({
       ? getLabelStyle(hasError)
       : cx(visuallyHidden);
     return (
-      <legend className={cx(legendReset, legendClassName)}>
+      <legend
+        className={cx(legendReset, legendClassName)}
+        hidden={!showListLabel}
```
Comment (nataliepina) [inline packages/toggleInputList/components/ToggleInputList.tsx:103]: What required the addition of `hidden`? Is it needed in conjunction with this class that applies the visually hidden styles?
```
const legendClassName = showListLabel
      ? getLabelStyle(hasError)
      : cx(visuallyHidden);
```

File: packages/toggleInputList/components/ToggleInputList.tsx [hunk 3]
```diff
@@ -140,7 +143,7 @@ const ToggleInputList = ({
       };
 
       return (
-        <li className={padding("top", "xs")} key={id}>
+        <li className={padding("top", "xs")} key={itemId}>
```
Comment (nataliepina) [inline packages/toggleInputList/components/ToggleInputList.tsx:146]: Why did you need to set the key to `itemId`?
It is cast above in the destructured props so that id = itemId like so:
`id: itemId,`

File: packages/toggleInputList/tests/ToggleInputList.test.tsx (hunk 1 above)
Comment (clintonmedbery) [reply packages/toggleInputList/tests/ToggleInputList.test.tsx:95]: It fails if it does not find the element.

**Review by clintonmedbery — COMMENTED** [review]
Thanks! Big improvements. I had the same question as Natalie on the hidden prop.

File: packages/toggleInputList/tests/ToggleInputList.test.tsx (hunk 1 above)
Comment (nataliepina) [reply packages/toggleInputList/tests/ToggleInputList.test.tsx:95]: Couldn't this also check for the required prop also given - `"renders with required"`? The danger test existing does confirm but I wonder if it could be more specific.

File: packages/toggleInputList/components/ToggleInputList.tsx (hunk 2 above)
Comment (taksuparth) [reply packages/toggleInputList/components/ToggleInputList.tsx:103]: @nataliepina @clintonmedbery according to the current implementation, when we want to hide the label, we use CSS 
```
clip: rect(0 0 0 0)
```
Due to this, elements are still present in the DOM structure and we cannot test 
```
expect(queriedElement).not.toBeInTheDocument();
```
Also, we cannot use 
```
expect(queriedElement).not.toBeVisible();
```
Because according to the react testing library documentation https://github.com/testing-library/jest-dom#tobevisible

> An element is visible if all the following conditions are met:
> 
> - it is present in the document
> - it does not have its css property display set to none
> - it does not have its css property visibility set to either hidden or collapse
> - it does not have its css property opacity set to 0
> - its parent element is also visible (and so on up to the top of the DOM tree)
> - it does not have the [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute
> - if it has the open attribute

And according to MDN docs hidden attribute is used when we want to hide a specific part of the DOM without removing it from DOM. So, expecting that is what required here added this hidden flag.

File: packages/toggleInputList/tests/ToggleInputList.test.tsx (hunk 1 above)
Comment (taksuparth) [reply packages/toggleInputList/tests/ToggleInputList.test.tsx:95]: @nataliepina didn't get you. With react testing library, we intend to test what the user perceives. So, I don't think we should test any prop. So, ideally here we should check for asteric sign for the required field. But it was getting difficult to find svg. So, instead of finding SVG tested with it's test-id. Do let me know if there is any action item on this thread.

**Review by nataliepina — APPROVED** [review]
